Bookish/Controllers/CatalogController.cs

```diff
@@ -49,41 +59,161 @@ public IActionResult BookEntry(BookInputModel model)
                 context.SaveChanges();
-                AddXCopies(newCopies, context, book);
+                AddCopiesOfBook(newCopies, context, book);
+            }
+        }
+        return View();
+    }
+    
+    public IActionResult AddNewBorrower(BookOrBorrowerInputModel model)
+    {
+        string surname = model.BorrowerInput.Surname;
+        string forename = model.BorrowerInput.Forename;
+        string phonenumber = model.BorrowerInput.PhoneNumber;
+        using (var context = new LibraryContext())
+        {
+            Borrower? foundBorrower = context.Borrowers.SingleOrDefault(a => a.Surname == surname && a.Forename == forename && a.PhoneNumber == phonenumber );
+            if (foundBorrower != null)
+            {
+                //member already exists, do something
+            }
+            else
+            {
+                var borrower = new Borrower()
+                {
+                    Surname = surname,
+                    Forename = forename,
+                    PhoneNumber = phonenumber
+                };
+                context.Borrowers.Add(borrower);
+                context.SaveChanges();
             }
         }
         return View();
     }
 
-    public void AddXCopies(int newCopies, LibraryContext context, Book book)
+    public void AddCopiesOfBook(int newCopies, LibraryContext context, Book book)
     {
+        List<Copy> newCopyList = new List<Copy>();
         for (int i = 0; i < newCopies; i++)
         {
             var copy = new Copy()
             {
                 BookID = book.BookID,
                 Comments = "This is a comment"
             };
-            context.Copies.Add(copy);
-            context.SaveChanges();
+            newCopyList.Add(copy);
         }
+        context.Copies.AddRange(newCopyList);
+        context.SaveChanges();
     }
-    
+    [HttpGet]
     public IActionResult BookList()
     {
         BookViewModel bvm = new BookViewModel();
         using (var context = new LibraryContext())
         {
-            bvm.EntryQuery = context.Books.Include(b => b.Author)
-                .Include(b => b.CopyList);
-            bvm.CatalogEntries = bvm.EntryQuery.ToList();
+            bvm.CatalogEntries = context.Books.Include(b => b.Author)
+                .Include(b => b.CopyList)
+                .ToList();
         }
-        return View(bvm);
+        return View("BookList",bvm);
     }
+    /*[HttpPost]
+    public IActionResult BookList(int bookId, string operation)
+    {
+        switch (operation)
+        {
+            case "delete":
+                throw new Exception("HELP DELETE");
+                return Delete(new BookViewModel());
+                break;
+            case "edit":
+                throw new Exception("HELP EDIT");
+                return Edit(new BookViewModel());
+                break;
+            default:
+                return View("Error");
+        }
+    }*/
+    [HttpGet]

```

Comment:
I appreciate this probably ought to be a Post or possibly a Put (because if you sent it multiple times it shouldn't change anything?) it is currently functional, and ideally would lead to an "Are you sure" page, which will be worked on as a new feature after this merge.